cardano-cli/src/Cardano/CLI/Commands/Governance.hs

```diff
@@ -31,41 +32,34 @@ runGovernanceCreateVoteCmd
 
-    VDR -> do
-      votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe vStakeCred
-      let voter = VoterDRep votingCred
-          govActIdentifier = shelleyBasedEraConstraints sbe $ createGovernanceActionId govActionTxId govActionIndex
-          voteProcedure = createVotingProcedure sbe vChoice Nothing
-          votingEntry = VotingEntry { votingEntryVoter = voter
-                                    , votingEntryGovActionId = GovernanceActionId govActIdentifier
-                                    , votingEntryVotingProcedure = voteProcedure
-                                    }
-      firstExceptT WriteFileError . newExceptT $ shelleyBasedEraConstraints sbe $ writeFileTextEnvelope oFp Nothing votingEntry
+  shelleyBasedEraConstraints sbe $ do
+    vStakePoolKey <- firstExceptT ReadFileError . newExceptT $ readVerificationKeyOrFile AsStakePoolKey votingStakeCred
+    let stakePoolKeyHash = verificationKeyHash vStakePoolKey
+        vStakeCred = StakeCredentialByKey . (verificationKeyHash . castVerificationKey) $ vStakePoolKey
+    case vType of
+      VCC -> do
+        votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe vStakeCred
+        let voter = Ledger.CommitteeVoter (Ledger.coerceKeyRole (unVotingCredential votingCred)) -- TODO Conway - remove coerceKeyRole
+            govActIdentifier = createGovernanceActionId govActionTxId govActionIndex
+            voteProcedure = createVotingProcedure sbe vChoice Nothing
+            votingProcedures = singletonVotingProcedures sbe voter govActIdentifier (unVotingProcedure voteProcedure)
+        firstExceptT WriteFileError . newExceptT $ writeFileTextEnvelope oFp Nothing votingProcedures
 
-    VSP -> do
-      let voter = VoterSpo stakePoolKeyHash
-          govActIdentifier = shelleyBasedEraConstraints sbe $ createGovernanceActionId govActionTxId govActionIndex
-          voteProcedure = createVotingProcedure sbe vChoice Nothing
-          votingEntry = VotingEntry { votingEntryVoter = voter
-                                    , votingEntryGovActionId = GovernanceActionId govActIdentifier
-                                    , votingEntryVotingProcedure = voteProcedure
-                                    }
-      firstExceptT WriteFileError . newExceptT $ shelleyBasedEraConstraints sbe $ writeFileTextEnvelope oFp Nothing votingEntry
+      VDR -> do
+        votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe vStakeCred
+        let voter = Ledger.DRepVoter (unVotingCredential votingCred)
+            govActIdentifier = createGovernanceActionId govActionTxId govActionIndex
+            voteProcedure = createVotingProcedure sbe vChoice Nothing
+            votingProcedures = singletonVotingProcedures sbe voter govActIdentifier (unVotingProcedure voteProcedure)
```

Comment:
Not for this PR but there is no reason why we shouldn't allow multiple voting files to be added to a tx.